python/ray/data/_internal/execution/operators/actor_pool_map_operator.py

```diff
@@ -66,7 +66,9 @@ def __init__(
             ObjectRef[ObjectRefGenerator], Tuple[_TaskState, ray.actor.ActorHandle]
         ] = {}
         # A pool of running actors on which we can execute mapper tasks.
-        self._actor_pool = _ActorPool(autoscaling_policy._config.max_tasks_in_flight)
+        # We start with 1 task in flight, then ramp up to `max_tasks_in_flight`
```

Comment:
How about implementing this ramp-up behavior within ActorPool itself? It can avoid dispatching multiple tasks when internally it has .

This would also be easier to unit test, as we already have actor pool unit tests in test_actor_pool_map_operator.py

Reply:
yeah good point...will make the change

Reply:
updated and added unit test